src/src/WebServer/JSON.cpp

```diff
@@ -448,8 +448,11 @@ void handle_json()
         stream_next_json_object_value(F("Type"),             getPluginNameFromDeviceIndex(DeviceIndex));
         stream_next_json_object_value(F("TaskName"),         getTaskDeviceName(TaskIndex));
         stream_next_json_object_value(F("TaskDeviceNumber"), Settings.TaskDeviceNumber[TaskIndex]);
-        if (Settings.TaskDeviceNumber[TaskIndex] == 1) {
-          stream_next_json_object_value(F("TaskDeviceGPIO"), Settings.TaskDevicePin1[TaskIndex]);
+        for(int i = 0; i < 3; i++)
+        {
+          if (Settings.TaskDevicePin[i][TaskIndex] != -1) {
+    	      stream_next_json_object_value(F("TaskDeviceGPIO") , Settings.TaskDevicePin[i][TaskIndex]);
```

Comment:
This will show 3x the same JSON label.
Maybe use `concat(F("TaskDeviceGPIO"), i)`
This may need `#include "../Helpers/StringConverter.h" ` (not sure about the exact name, you may need to check it)